File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        count = 0
        while len(set(nums)) != len(nums):
            if len(nums) < 3:
                nums = []
            else:
                nums = nums[3:]
            count += 1
        return count 

File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.ts
function minimumOperations(nums: number[]): number {
    const hashMap = new Array<boolean>(101).fill(false);
    let firstDuplicateIdx = nums.length;

    for (let i = nums.length - 1; i >= 0; i--) {
        if (hashMap[nums[i]]) break;

        hashMap[nums[i]] = true;
        firstDuplicateIdx = i;
    }
    return ((firstDuplicateIdx + 2) / 3) | 0;
};